openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -946,12 +946,12 @@ If the Credential Response is not encrypted, the media type of the response MUST
 
 The following parameters are used in the JSON-encoded Credential Response body:
 
-* `credential`: OPTIONAL. Contains issued Credential. It MUST NOT be used if `credentials` or `transaction_id` parameter is present. It MAY be a string or an object, depending on the Credential Format. See (#format-profiles) for the Credential Format-specific encoding requirements.
-* `credentials`: OPTIONAL. Contains an array of issued Credentials. It MUST NOT be used if `credential` or `transaction_id` parameter is present. The values in the array MAY be a string or an object, depending on the Credential Format. See (#format-profiles) for the Credential Format-specific encoding requirements.
-* `transaction_id`: OPTIONAL. String identifying a Deferred Issuance transaction. This claim is contained in the response if the Credential Issuer cannot immediately issue the Credential. The value is subsequently used to obtain the respective Credential with the Deferred Credential Endpoint (see (#deferred-credential-issuance)). It MUST not be used  if `credential` or `credentials` is present. It MUST be invalidated after the Credential for which it was meant has been obtained by the Wallet.
+* `credentials`: OPTIONAL. Contains an array of one or more issued Credentials. It MUST NOT be used if the `transaction_id` parameter is present. The elements of the array MUST be objects. The following parameters are used inside this object (other parameters MAY also be included):
+   * `credential`: REQUIRED. Contains one issued Credential. It MAY be a string or an object, depending on the Credential Format. See Appendix A for the Credential Format-specific encoding requirements.
+   * `notification_id`: OPTIONAL. String identifying an issued Credential that the Wallet includes in the Notification Request as defined in (#notification).
```

Comment:
Shouldn't we have a single `notification_id` for the whole response?

Reply:
I've been struggling with the same question for a few days:

A `notification_id` for each credential in the batch is cleaner/simpler, as we don't need to worry about partial states (i.e. the question of what notification to make if you receive 10 credentials and only manage to store 5 of them).

However it's kindof sucky to make 10 notification http calls for a batch of 10 credentials.

I'm probably veering towards cleaner/simpler and ignoring the 'kind of sucky' part.

Reply:
We may change the notification request to an array as well?

Reply:
I think that could make a lot of sense.

Reply:
FWIW, I think that `notification_id` should point to a single credential not in a batch.

For instance, an Issuer can react to a notification event carrying a `credential_deleted` or `credential_failure` to invalidate the credential from a status list.

We can avoid multiple posts for notifications by using an array of `notification_id` per possible event.

```json
{
  "credential_accepted" : [ "notId#1"],
  "credential_failure" : ["notid#4"],
  "credential_deteled": [ ]
}
```

Something like a summary notification

Reply:
I think it might be useful to move it in because of `credential_deleted` which is specific to one issued credential although when I think about it, `notification_id` overlaps a bit with the concept of credential version IDs or data set IDs but this is out of scope of this PR.

Reply:
Why would a wallet delete a single instance rather than the whole batch or the same Credential Dataset?

Reply:
maybe because the user has used that one instance? but not sure that information needs to be communicated to the issuer real-time, since that would allow the issuer to get more information about when credential is used